### PR TITLE
Refactor TX list and details

### PR DESCRIPTION
**Pull request by jpalvarezl**
Handles #814 #1055 

Changes proposed in this pull request:
- Flattened models in the app
- Tx list and tx details use models directly from the gateway
- Tests were left as untouched as possible

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt [hunk 1]
```diff
@@ -591,5 +568,8 @@ fun TransactionView.isHistory() = when (status) {
 fun Solidity.Address.getVersionForAddress(): String = masterCopyVersion(this) ?: SAFE_MASTER_COPY_UNKNOWN_DISPLAY_STRING
 
 fun Transaction.canBeSignedByOwner(ownerAddress: Solidity.Address?): Boolean {
-    return missingSigners?.contains(ownerAddress) == true
+    return executionInfo?.missingSigners?.contains(ownerAddress) == true
 }
+
+//TODO add unit tests
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt:574]: Add tests

**Review by elgatovital — APPROVED** [review]
LGTM! 👍 Please create an issue for moving decoded param classes from backend.dto.TransactionDto to models.transaction

**Comment by jpalvarezl** [comment]
I opened the issue for Android: https://github.com/gnosis/safe-android/issues/1070

and for the gateway: https://github.com/gnosis/safe-client-gateway/issues/190

**Comment by elgatovital** [comment]
image
same for item_tx_change_mastercopy, etc. All tx item layouts should be checked and adjusted

**Comment by jpalvarezl** [comment]
What is wrong with that one?

**Comment by elgatovital** [comment]
it has to be right aligned

**Comment by jpalvarezl** [comment]
You are right, the `viewEnd` was the problem just like with `textEnd`. It was only for the master copy as far as I could see.

**Pull request merged**